clipper_admin_v2/clipper_admin/container_manager.py

```diff
@@ -0,0 +1,87 @@
+import abc
+
+# Constants
+
+CLIPPER_QUERY_PORT = 1337
+CLIPPER_MANAGEMENT_PORT = 1338
+CLIPPER_RPC_PORT = 7000
+
+CLIPPER_DOCKER_LABEL = "ai.clipper.container.label"
+CLIPPER_MODEL_CONTAINER_LABEL = "ai.clipper.model_container.label"
+CONTAINERLESS_MODEL_IMAGE = "NO_CONTAINER"
+
+
+class ContainerManager(object):
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, clipper_public_hostname):
+        self.public_hostname = clipper_public_hostname
+
+    @abc.abstractmethod
+    def start_clipper(self):
```

Comment:
Seems strange that the `cm` is responsible for starting clipper, but `cm`'s constructor takes Clipper hostname as an argument (which isn't known until after Clipper is started).

I would've made`start_clipper` concrete and have it delegate to an abstract `_start_clipper` that is overridden by implementations of the abc which returns the hostname, then have `start_clipper` (this method) set `self.public_hostname` with the return.